src/components/Chat/Chat.tsx

```diff
@@ -12,23 +12,58 @@ interface ChatProps {
   messages: OnMessageCallbackData[];
   onSendMessage(text: string, image?: File): any;
   onNewQuestionClick(): any;
+  index: number;
+  allChats: any;
+  onCheckmarkClick(): any;
 }
 
 export default function Chat(props: ChatProps) {
   const { messages } = props;
 
+  const emptyCheckmark = (
+    <button
+      id="chat-empty-checkmark"
+      className="profile-item-emptycheckmark"
+      onClick={props.onCheckmarkClick}
+    />
+  );
+  const checkedCheckmark = (
+    <button id="chat-checkmark" className="profile-item-checkmark" />
+  );
+  function displayCheckmark(bool) {
```

Comment:
It's generally a bad idea to have the data type as the variable name, it should be something like "checked" instead.

Reply:
It's mostly for clarity reasons.